mongodb/mongodbatlas-cloudformation-resources
# INTMDB-475 Cloud backup snapshot export bucket

## Description

Added new Resource Cloud Backup export bucket:

The exportBuckets resource allows you to grant Atlas access to the specified bucket for exporting backup snapshots.

How to test:
In order to test this resource son Prerequisites are required, check [prerequisites](https://www.mongodb.com/docs/atlas/backup/cloud-backup/export/#prerequisites)

1. M10 or higher Atlas cluster with Cloud Backup [enabled.](https://www.mongodb.com/docs/atlas/backup/cloud-backup/overview/#std-label-backup-cloud-provider)

2. AWS [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html) with STS:AssumeRole that grants Atlas access to your AWS resources. To learn more about configuring AWS access for Atlas, see [Set Up Unified AWS Access.](https://www.mongodb.com/docs/atlas/security/set-up-unified-aws-access/#std-label-set-up-unified-aws-access)

3. AWS [IAM role policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) that grants Atlas write access or the S3:PutObject and S3:GetBucketLocation permissions to your AWS resources. To learn more about configuring write access to AWS resources, see [Set Up Unified AWS Access.](https://www.mongodb.com/docs/atlas/security/set-up-unified-aws-access/#std-label-set-up-unified-aws-access)

Two main resource are required:

1. an Atlas and a Aws role related by a trusted relationship, and permissions to access an s3 bucket
2. an s3 bucket (the bucket name is required)

<img width="1617" alt="Screenshot 2022-11-25 at 8 25 09 AM" src="https://user-images.githubusercontent.com/69170650/203995149-40ec2aca-e205-4ef0-b4fd-5ea95f48f8e9.png">

Can be manually executed by using this json inputs and the command:


`cfn invoke resource {CREATE | READ | UPDATE (NOT SUPPORTED) | DELETE | LIST} file-name.json`
    
CREATE
```json
{
  "desiredResourceState": {
    {
  "GroupId": {GROUP-ID},
  "IamRoleID": {ROLE-ID-CREATED},
  "BucketName": {AWS-S3-BUCKET-NAME}
  "ApiKeys": {
    "PublicKey": {PUBLIC-KEY},
    "PrivateKey": {PRIVATE-KEY}
  }
}
  },
  "previousResourceState": {
  }
}
```
    
READ
```json
{
  "desiredResourceState": {
    {
  "Id": {EXPORT-BUCKET-ID},
  "GroupId": {GROUP-ID},
  "IamRoleID": {ROLE-ID-CREATED},
  "BucketName": {AWS-S3-BUCKET-NAME}
  "ApiKeys": {
    "PublicKey": {PUBLIC-KEY},
    "PrivateKey": {PRIVATE-KEY}
  }
}
  },
  "previousResourceState": {
  }
}
```

DELETE
```json
{
  "desiredResourceState": {
    {
  "Id": {EXPORT-BUCKET-ID},
  "GroupId": {GROUP-ID},
  "IamRoleID": {ROLE-ID-CREATED},
  "BucketName": {AWS-S3-BUCKET-NAME}
  "ApiKeys": {
    "PublicKey": {PUBLIC-KEY},
    "PrivateKey": {PRIVATE-KEY}
  }
}
  },
  "previousResourceState": {
  }
}
```

LIST
```json
{
  "desiredResourceState": {
    {
  "GroupId": {GROUP-ID},
  "IamRoleID": {ROLE-ID-CREATED},
  "BucketName": {AWS-S3-BUCKET-NAME}
  "ApiKeys": {
    "PublicKey": {PUBLIC-KEY},
    "PrivateKey": {PRIVATE-KEY}
  }
}
  },
  "previousResourceState": {
  }
}
```

Link to any related issue(s): 

## Type of change:


## Required Checklist:

- [x] I have added any necessary documentation (if appropriate)

## Further comments
